imports/plugins/core/accounts/server/no-meteor/mutations/inviteShopMember.js

```diff
@@ -0,0 +1,140 @@
+    throw new ReactionError("bad-request", "Cannot directly invite owner");
+  }
+
+  // check to see if invited user has an account
+  const invitedUser = await users.findOne({ "emails.address": email });
+
+  // if there is a userAccount, check to see if email has been verified
+  const matchingEmail = invitedUser && invitedUser.emails && invitedUser.emails.find((accountEmail) => accountEmail.address === email);
+  const isEmailVerified = matchingEmail && matchingEmail.verified;
+
+  // set variables to pass to email templates
+  const invitedByName = getCurrentUserName(userFromContext);
+  let dataForEmail;
+  let userId;
+  let templateName;
+
+  if (invitedUser) {
+    userId = invitedUser._id;
+  }
+
+  // if the user already has an account, send in informative email instead of an invite email
+  if (invitedUser && isEmailVerified) {
+    // make sure user has an `Account` - this should always be the case
+    const invitedAccount = await Accounts.findOne({ userId });
+    if (!invitedAccount) throw new ReactionError("not-found", "User found but matching account not found");
+
+    // do not send token, as no password reset is needed
+    const url = context.getAbsoluteUrl();
+
+    // use primaryShop's data (name, address etc) in email copy sent to new shop manager
+    dataForEmail = getDataForEmail(context, { shop: primaryShop, currentUserName: invitedByName, name, url });
+
+    // Get email template and subject
+    templateName = "accounts/inviteShopMember";
+  } else {
+    // There could be an existing user with an invite still pending (not activated).
+    // We create a new account only if there's no pending invite.
+    if (!invitedUser) {
+      // The user does not already exist, we need to create a new account
+      userId = await context.createUser({
+        profile: { invited: true },
+        email,
+        name,
+        groupId
+      });
+    }
+
+    const token = Random.id();
+
+    // set token to be used for first login for the new account
+    const tokenUpdate = {
+      "services.password.reset": { token, email, when: new Date() },
+      name
+    };
+
+    await users.updateOne({ _id: userId }, { $set: tokenUpdate }); // TODO: not sure if this is working
+
+    // use primaryShop's data (name, address etc) in email copy sent to new shop manager
+    dataForEmail = getDataForEmail(context, { shop: primaryShop, currentUserName: invitedByName, name, token });
+
+    // Get email template and subject
+    templateName = "accounts/inviteNewShopMember";
+  }
+
+  // add new / existing user to invited group
+  await context.mutations.addAccountToGroup(context, {
+    accountId: userId,
+    groupId
+  });
+
+  dataForEmail.groupName = _.startCase(group.name);
+
+  // send invitation email from primary shop email
+  await context.mutations.sendEmail(context, {
+    data: dataForEmail,
+    fromShop: primaryShop,
+    templateName,
+    to: email
+  });
+
+  return Accounts.findOne({ _id: userId });
```

Comment:
Do `Accounts.findOne({ userId });` same as above rather than relying on `_id` being the same as `userId`.

Reply:
Not sure if you saw my comment on that comment above. I think `userId` just might be a badly named `var`, because `userId` actually is just `_id` field.

https://github.com/reactioncommerce/reaction/pull/5520#discussion_r321074146

If you still think it should be changed, I will, but maybe just the `var` name should be different instead?